### ISSUE TITLE
NPE

### ISSUE THREAD

**Issue by integrii** [Comcast/kuberhealthy#144]
```
time="2019-03-18T07:48:18Z" level=info msg="DaemonSetChecker Daemonset check waiting for 1 pods to come up on nodes [ip-10-32-14-229.eu-west-1.compute.internal]"
time="2019-03-18T07:48:19Z" level=info msg="DaemonSetChecker Daemonset check waiting for 1 pods to come up on nodes [ip-10-32-14-229.eu-west-1.compute.internal]"
time="2019-03-18T07:48:21Z" level=info msg="DaemonSetChecker Daemonset check waiting for 1 pods to come up on nodes [ip-10-32-14-229.eu-west-1.compute.internal]"
time="2019-03-18T07:48:25Z" level=info msg="DaemonSetChecker Daemonset check waiting for 1 pods to come up on nodes [ip-10-32-14-229.eu-west-1.compute.internal]"
time="2019-03-18T07:48:38Z" level=info msg="Client connected to status page from 100.107.226.192:22004 Go-http-client/1.1"
time="2019-03-18T07:48:45Z" level=info msg="Running check: ComponentStatusChecker"
time="2019-03-18T07:48:45Z" level=info msg="Running check: PodStatusChecker namespace kube-system"
time="2019-03-18T07:49:10Z" level=info msg="Client connected to status page from 10.32.14.160:41584 Go-http-client/1.1"
time="2019-03-18T07:49:12Z" level=info msg="Client connected to status page from 100.107.159.0:16156 python-requests/2.18.4"
time="2019-03-18T07:49:40Z" level=info msg="Client connected to status page from 10.32.14.210:36467 Go-http-client/1.1"
time="2019-03-18T07:49:51Z" level=info msg="Client connected to status page from 10.32.14.234:53050 Go-http-client/1.1"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xf77cb1]

goroutine 27 [running]:
main.(*Kuberhealthy).setCheckExecutionError(0xc00010e0f0, 0xc0005d1110, 0x26, 0x0, 0x0)
	/go/src/github.com/Comcast/kuberhealthy/cmd/kuberhealthy/kuberhealthy.go:63 +0x211
main.(*Kuberhealthy).runCheck(0xc00010e0f0, 0xc0002e0230, 0x1302fe0, 0xc0002bc840)
	/go/src/github.com/Comcast/kuberhealthy/cmd/kuberhealthy/kuberhealthy.go:242 +0x366
created by main.(*Kuberhealthy).StartChecks
	/go/src/github.com/Comcast/kuberhealthy/cmd/kuberhealthy/kuberhealthy.go:123 +0xf9
```

**Comment by integrii** [comment]
Interesting result here.  The pod crashed with daemonset checkers on, then restarted.  When restarting, the pod maintained the same name.  That meant the pod was still "master".  That also meant that the pods left from the crashed instance were NOT identified as orphaned and were allowed to lay around.

**Comment by integrii** [comment]
Also, the pods kept re-creating because an orphaned `daemonset` was hanging around spinning up pods that would never get deleted and was unbeknownst to the previously crashed instance.  We need to label daemonset pods and `daemonset` instance better so that better orphan checks can be run.

**Comment by integrii** [comment]
See #144

**Comment by lolimjake** [comment]
This appears to be a duplicate of #114 - which is fixed in v1.0.1 +.  Your related issue #145 is valid, though.
